include/aspect/postprocess/particle_crystal_preferred_orientation.h

```diff
@@ -0,0 +1,322 @@
+
+ This file is part of ASPECT.
+
+ ASPECT is free software; you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation; either version 2, or (at your option)
+ any later version.
+
+ ASPECT is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with ASPECT; see the file LICENSE.  If not see
+ <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef _aspect_postprocess_particle_crystal_preferred_orientation_h
+#define _aspect_postprocess_particle_crystal_preferred_orientation_h
+
+#include <aspect/postprocess/interface.h>
+#include <aspect/particle/world.h>
+
+#include <aspect/simulator_access.h>
+
+#include <deal.II/particles/particle_handler.h>
+#include <deal.II/base/data_out_base.h>
+#include <tuple>
+
+namespace aspect
+{
+  namespace Postprocess
+  {
+    /**
+     * A Postprocessor that creates particles, which follow the
+     * velocity field of the simulation. The particles can be generated
+     * and propagated in various ways and they can carry a number of
+     * constant or time-varying properties. The postprocessor can write
+     * output positions and properties of all particles at chosen intervals,
+     * although this is not mandatory. It also allows other parts of the
+     * code to query the particles for information.
+     */
+    template <int dim>
+    class ParticleCrystalPreferredOrientation : public Interface<dim>, public ::aspect::SimulatorAccess<dim>
+    {
+      public:
+        /**
+         * Constructor.
+         */
+        ParticleCrystalPreferredOrientation();
+
+        /**
+         * Destructor.
+         */
+        ~ParticleCrystalPreferredOrientation();
+
+
+        /**
+         * Initialize function.
+         */
+        virtual void initialize ();
+
+        /**
+         * Execute this postprocessor. Derived classes will implement this
+         * function to do whatever they want to do to evaluate the solution at
+         * the current time step.
+         *
+         * @param[in,out] statistics An object that contains statistics that
+         * are collected throughout the simulation and that will be written to
+         * an output file at the end of each time step. Postprocessors may
+         * deposit data in these tables for later visualization or further
+         * processing.
+         *
+         * @return A pair of strings that will be printed to the screen after
+         * running the postprocessor in two columns; typically the first
+         * column contains a description of what the data is and the second
+         * contains a numerical value of this data. If there is nothing to
+         * print, simply return two empty strings.
```

Comment:
Update this description to be specific for this postprocessor, not the general one from the interface.